kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/CaseInsensitiveMap.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A map that is case-insensitive in its keys. This map is not thread-safe, and key is
+ * case-insensitive and of string type.
+ *
+ * @param <V>
+ */
+public class CaseInsensitiveMap<V> implements Map<String, V> {
```

Comment:
Would it make sense to call it something like `CaseInsensitiveStringMap` since the key is always a string?